### PR TITLE
[BREAKINGCHANGE]: move calculation model to `core` package AND s/lodash-es/lodash

### PR DESCRIPTION
**Pull request by juliepagano**
This change also replaces lodash-es with lodash, which should be a noop for most consumers, but will address some ESM-only issues in certain use cases (e.g. next.js).

BREAKINGCHANGE: calculation-related utilities and types have moved from the `plugin-system` package to the `core` package and associated imports will need to be adjusted.

# Notes for reviewers

- Doing this as part of responding to https://github.com/perses/perses/pull/1237#discussion_r1222288084. Doing the move of the utility to core in a separate PR, so any changes I make to combine the logic are a little easier to see in a git diff.
- This PR surfaced the esm-only issue from lodash-es again, so I'm fixing it while I'm in here.

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Review by sjcobb — APPROVED** [review]
🙏

File: ui/components/src/EChart/EChart.tsx [hunk 1]
```diff
@@ -14,7 +14,8 @@
 import React, { useEffect, useLayoutEffect, useRef } from 'react';
 import { ECharts, EChartsCoreOption, init, connect } from 'echarts/core';
 import { Box, SxProps, Theme } from '@mui/material';
-import { isEqual, debounce } from 'lodash-es';
+import isEqual from 'lodash/isEqual';
+import debounce from 'lodash/debounce';
```
Comment (juliepagano) [inline ui/components/src/EChart/EChart.tsx:18]: I _think_ you don't need the nested imports in all use cases (e.g. I think webpack may be smart enough these days to tree shake with normal imports), but using them to be extra safe.

Comment (sjcobb) [reply ui/components/src/EChart/EChart.tsx:18]: Yeah I'm still don't understand what the "correct" way to use lodash is, this approach is what I landed on [in core too](https://github.com/perses/perses/blob/v0.33.0/ui/core/src/utils/memo.ts#L15) when I needed the fix for e2e's in #1165

File: ui/jest.shared.ts [hunk 2]
```diff
@@ -23,8 +23,6 @@ const config: Config.InitialOptions = {
   testEnvironment: 'jsdom',
   roots: ['<rootDir>/src'],
   moduleNameMapper: {
-    // Jest currently doesn't natively support ES Modules, so use the non ES Modules versions instead
-    '^lodash-es$': 'lodash',
```
Comment (sjcobb) [inline ui/jest.shared.ts:27]: 🎉

**Review by sjcobb — APPROVED** [review]
Lodash changes LGTM to me!

**Pull request merged**